### PR TITLE
Adding Week 3 BlogPost

### PR DESCRIPTION
**Pull request by ganimtron-10**
Adding Week 3 Blog Post

BlogPost Week3

### CODE REVIEW

**Review by m-agour — COMMENTED** [review]
Hello @ganimtron-10 ,
Nice and detailed post with no typos I believe.
I only have some minor punctuation related suggestions.

File: docs/source/posts/2022/2022-06-22-week-3-praneeth.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    :align: center
+
+But then the question arises `How to efficiently take data from the user?`, `Which data format would be easy to compute and best for user experience?` and so on.
+
+Alongside I was trying to create polylines but whenever I wanted to start the creation of the second line the dragging event wasn't triggered as each dragging event required a left mouse click event associated with it. 
+I tried to manually do that but it didn't work.
+
+Did you get stuck anywhere?
+---------------------------
+As I mentioned when I translate the shapes it would go out of the canvas bounds, here the problem was with the reference point by which I was calculating all the transformations it changed depending on various cases as shown below.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:29]: Missing full stop before `here the problem`

File: docs/source/posts/2022/2022-06-22-week-3-praneeth.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   :tags: google
+   :category: gsoc
+
+
+What did you do this week?
+--------------------------
+This week was full of researching, playing around with things, prototyping ideas, etc.
+I started with last week's clamping issue and managed to solve this issue while drawing shapes by clipping the mouse position according to canvas size but it didn't solve the problem with translating these shapes. I tried solving this using various approaches but if one thing would get fixed other things would raise a new error.
+
+Instead of spending too much time on this, I thought it would be great to switch to other work and discuss this problem with the mentors. So I started to create a basic prototype for the properties panel which would be used to manipulate different properties of the selected shape.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:16]: Missing a comma after `properties panel`

File: docs/source/posts/2022/2022-06-22-week-3-praneeth.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+.. post:: June 22 2022
+   :author: Praneeth Shetty 
+   :tags: google
+   :category: gsoc
+
+
+What did you do this week?
+--------------------------
+This week was full of researching, playing around with things, prototyping ideas, etc.
+I started with last week's clamping issue and managed to solve this issue while drawing shapes by clipping the mouse position according to canvas size but it didn't solve the problem with translating these shapes. I tried solving this using various approaches but if one thing would get fixed other things would raise a new error.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:14]: Missing a comma after `would get fixed`

Comment (m-agour) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:14]: Missing a comma after `using various approaches`

Comment (m-agour) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:14]: Missing a comma after `canvas size`

**Comment by ganimtron-10** [comment]
Thanks, @m-agour for the review!

File: docs/source/posts/2022/2022-06-22-week-3-praneeth.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Instead of spending too much time on this, I thought it would be great to switch to other work and discuss this problem with the mentors. So I started to create a basic prototype for the properties panel, which would be used to manipulate different properties of the selected shape.
+
+.. image:: https://user-images.githubusercontent.com/64432063/176094716-6be012b8-36c5-43e7-a981-612dbd37ab09.gif
+    :width: 300
+    :align: center
+
+But then the question arises `How to efficiently take data from the user?`, `Which data format would be easy to compute and best for user experience?` and so on.
+
+Alongside I was trying to create polylines but whenever I wanted to start the creation of the second line the dragging event wasn't triggered as each dragging event required a left mouse click event associated with it. 
+I tried to manually do that but it didn't work.
```
Comment (xtanion) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:25]: `do that manually` instead

File: docs/source/posts/2022/2022-06-22-week-3-praneeth.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+Did you get stuck anywhere?
+---------------------------
+As I mentioned when I translate the shapes it would go out of the canvas bounds. Here the problem was with the reference point by which I was calculating all the transformations it changed depending on various cases as shown below.
```
Comment (xtanion) [inline docs/source/posts/2022/2022-06-22-week-3-praneeth.rst:29]: `translated` instead

**Review by xtanion — COMMENTED** [review]
Hi @ganimtron-10, This is a really nice blog. There are a few minor grammatical errors I mentioned below. PTAL

**Comment by ganimtron-10** [comment]
Thank you @xtanion  for the review!

**Review by skoudoro — APPROVED** [review]
Thank you @ganimtron-10, merging!

**Pull request merged**